telemetry/data_feature_usage.go

```diff
@@ -15,31 +15,60 @@ package telemetry
 
 import (
 	"context"
+	"errors"
 
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/infoschema"
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/store/tikv/metrics"
+	"github.com/pingcap/tidb/util/logutil"
 	"github.com/pingcap/tidb/util/sqlexec"
 )
 
 type featureUsage struct {
-	Txn          *TxnUsage       `json:"txn"`
-	ClusterIndex map[string]bool `json:"clusterIndex"`
+	// transaction usage information
+	Txn *TxnUsage `json:"txn"`
+	// cluster index usage information
+	// key is the first 6 characters of sha2(TABLE_NAME, 256)
+	ClusterIndex *ClusterIndexUsage `json:"clusterIndex"`
 }
 
 func getFeatureUsage(ctx sessionctx.Context) (*featureUsage, error) {
-	// init
-	usageInfo := featureUsage{
-		ClusterIndex: make(map[string]bool),
+
+	clusterIdxUsage, err := GetClusterIndexUsageInfo(ctx)
+	if err != nil {
+		logutil.BgLogger().Info(err.Error())
```

Comment:
How about continue reporting other features when the retrieval of some of the feature usage is failed?